src/server/pps/server/api_server.go

```diff
@@ -1619,15 +1619,21 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
 
 	// Delete output repo
 	if request.DeleteRepo {
-		pfsClient := pachClient.PfsAPIClient
-		if _, err := pfsClient.DeleteRepo(ctx, &pfs.DeleteRepoRequest{
-			Repo:  &pfs.Repo{request.Pipeline.Name},
-			Force: true,
-		}); err != nil {
+		var eg errgroup.Group
+		eg.Go(func() error {
+			return pachClient.WithCtx(ctx).DeleteRepo(request.Pipeline.Name, true)
+		})
+		pps.VisitInput(pipelineInfo.Input, func(input *pps.Input) {
+			if input.Cron != nil {
+				eg.Go(func() error {
+					return pachClient.WithCtx(ctx).DeleteRepo(input.Cron.Repo, true)
```

Comment:
It looks like we preserve Cron repos if `request.DeleteRepo == false`. Just checking: is that deliberate?

Reply:
Yes, my thinking is that `DeleteRepo == true` means, clean up all of the repos which were created as part of this pipeline, this includes the output repo and any cron repos but not the input repos because those existed before the pipeline was created.

Reply:
The point of leaving them around when `DeleteRepo == false` is that then you still have a full copy of the output data with provenance.